infer/resource.go

```diff
@@ -289,7 +365,7 @@ func (g *fieldGenerator) InputField(a any) InputField {
 }
 
 func (g *fieldGenerator) OutputField(a any) OutputField {
-	if allFields, ok, err := g.argsMatcher.TargetStructFields(a); ok {
+	if allFields, ok, err := g.stateMatcher.TargetStructFields(a); ok {
```

Comment:
This was a simple typo that testing missed. We didn't check for errors on the default case, and so we would generate an error on the default case always (and then ignore it). The assert on L835 prevents this from happening again.